code/modules/reagents/chemistry/reagents/water.dm

```diff
@@ -20,10 +20,65 @@
 	drink_desc = "The father of all refreshments."
 
 /datum/reagent/water/reaction_mob(mob/living/M, method=TOUCH, volume)
-// Put out fire
 	if(method == TOUCH)
+		// Put out fire
 		M.adjust_fire_stacks(-(volume / 10))
 		M.ExtinguishMob()
+		if(ishuman(M))
+
+			var/mob/living/carbon/human/H = M
+
+			if(H.species.name != "Grey") //God this is so gross I hate it.
```

Comment:
Actually, nevermind. `handle_reagents` doesn't seem to be able to determine the difference between ingest and surface application, which is kinda needed for the sacid-like reactions for water.